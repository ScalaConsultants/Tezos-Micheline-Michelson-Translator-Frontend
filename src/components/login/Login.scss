.login {
  .login-content {
    max-width: 500px;
    margin-left: auto;
    margin-right: auto;
    justify-content: center;
    .content__bottom-container {
      display: flex;
      width: 90%;
      .error-message {
        display: flex;
        justify-content: center;
        align-items: center;
        padding: 0 5px;
        margin: 0;
        width: 100%;
        font-family: Roboto;
        font-weight: 500;
        text-transform: uppercase;
        letter-spacing: 0.15;
        font-size: 14px;
        border-radius: 4px;
        border: 1px solid red;
        box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.2),
          0px 2px 2px rgba(0, 0, 0, 0.12), 0px 0px 2px rgba(0, 0, 0, 0.14);
        background: rgba(201, 0, 0, 0.5);
        color: white;
      }
    }
  }
}
