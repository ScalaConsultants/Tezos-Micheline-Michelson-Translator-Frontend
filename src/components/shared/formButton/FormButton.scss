@import "../../../scss/vars";

.form-button {
    font-weight: 500;
    text-transform: uppercase;
    letter-spacing: 0.15;
    font-size: 14px;
    box-shadow: $box-shadow;
    border-radius: 4px;
    padding: 5px;
    height: 40px;
    min-width: 100px;
    border: none;
    color: white;

    &:hover {
        cursor: pointer;
    }
}

.form-button_submit {
    background: $background-color-red;
}

.form-button_secondary {
    background: $background-color-grey-3;
}
