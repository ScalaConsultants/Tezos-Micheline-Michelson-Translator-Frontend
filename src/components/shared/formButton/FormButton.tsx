import React from "react";
import { ButtonProps } from "./types";

const FormButton = ({ type, label, className, disabled, stylingType, onClick }: ButtonProps) => (
  <button
    className={`form-button form-button_${stylingType || "submit"} ${className}`}
    data-testid="form-button"
    disabled={disabled}
    type={type}
    onClick={onClick}
  >
    {label}
  </button>
);

export default FormButton;
