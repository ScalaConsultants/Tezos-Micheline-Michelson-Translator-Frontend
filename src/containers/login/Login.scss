.login-page {
  .container_full {
    display: flex;
    width: 90%;
    margin-left: auto;
    margin-right: auto;

    @media screen and (max-width: 800px) {
      display: block;
    }
  }

  .sub-container_login-form {
    width: 65%;
    margin-left: auto;
    margin-right: auto;

    @media screen and (max-width: 800px) {
      width: 100%;
    }
  }

  .sub-container_login-info {
    width: 35%;

    @media screen and (max-width: 800px) {
      width: 80%;
      margin-left: auto;
      margin-right: auto;
      margin-bottom: 30px;
    }

    @media screen and (max-width: 500px) {
      width: 100%;
    }
  }
}
