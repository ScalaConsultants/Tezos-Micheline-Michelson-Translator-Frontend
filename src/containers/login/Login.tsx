import React from "react";
import LoginForm from "../../components/login/Login";
import "./Login.scss";

const Login = () => {
  return (
    <div className="login-page">
      <h1>Login</h1>
      <div className="container_full">
        <div className="sub-container_login-form">
          <LoginForm />
        </div>
      </div>
    </div>
  );
};

export default Login;
