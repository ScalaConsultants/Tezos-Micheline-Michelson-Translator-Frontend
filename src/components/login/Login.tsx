import React, { useEffect } from "react";
import { useRouter } from "next/router";
import { Formik } from "formik";
import * as Yup from "yup";
import { useMappedState, useDispatch } from "redux-react-hook";
import { bindActionCreators } from "redux";

import * as authTypes from "../../store/authentication/types";
import * as authActions from "../../store/authentication/actions";
import { IState } from "../../store/global/types";
import FormInput from "../shared/input/FormInput";
import FormButton from "../shared/formButton/FormButton";
import "./Login.scss";

const mapState = (state: IState) => ({
  auth: state.auth
});

const validationSchema = Yup.object().shape({
  username: Yup.string().required("Login is needed"),
  password: Yup.string().required("Password is needed")
});

const Login = () => {
  const { auth } = useMappedState(mapState);
  const dispatch = useDispatch();
  const router = useRouter();
  const boundAuthActions = bindActionCreators(authActions, dispatch);

  const submitForm = (values: authTypes.authCredentials) => {
    boundAuthActions.AuthenticationLogin(values);
  };

  useEffect(() => {
    auth.isLogged && router.push("/admin/library");
  }, [auth.isLogged]);

  return (
    <div className="login">
      <div className="login-content">
        <Formik
          initialValues={{
            username: "",
            password: ""
          }}
          validationSchema={validationSchema}
          onSubmit={submitForm}
        >
          {({
            values,
            errors,
            touched,
            handleChange,
            handleBlur,
            handleSubmit,
            isSubmitting
          }) => (
            <form onSubmit={handleSubmit} noValidate>
              <FormInput
                label="Login"
                type="text"
                name="username"
                onChange={handleChange}
                onBlur={handleBlur}
                value={values.username}
                errors={errors.username}
                touched={touched.username}
              />
              <FormInput
                label="Password"
                type="password"
                name="password"
                onChange={handleChange}
                onBlur={handleBlur}
                value={values.password}
                errors={errors.password}
                touched={touched.password}
              />
              <div className="content__bottom-container">
                <FormButton
                  label="Log in"
                  type="submit"
                  disabled={!!Object.keys(errors).length}
                />
                {isSubmitting && !auth.isError && (
                  <div className="error-message">
                    Incorrect login or password
                  </div>
                )}
              </div>
            </form>
          )}
        </Formik>
      </div>
    </div>
  );
};

export default Login;
