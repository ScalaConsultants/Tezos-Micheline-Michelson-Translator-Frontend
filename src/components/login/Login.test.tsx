import React from "react";
import { mount } from "enzyme/build";
import { StoreContext } from "redux-react-hook";
import Login from "./Login";
import Formbutton from "../shared/formButton/FormButton";

import configureStore from "../../store";
import rootSaga from "../../store/saga";
import FormButton from "../shared/formButton/FormButton";

const store = configureStore();
store.runSaga(rootSaga);

describe("Login", () => {
  const wrapper = mount(
    <StoreContext.Provider value={store}>
      <Login />
    </StoreContext.Provider>,
  );

  it("matches snapshot", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    const item = wrapper.find(Login);

    expect(
      item
        .at(0)
        .find({ name: "username" })
        .at(0)
        .text(),
    ).toEqual("Login");

    expect(
      item
        .at(0)
        .find(FormButton)
        .at(0)
        .text(),
    ).toEqual("Log in");
  });
});
