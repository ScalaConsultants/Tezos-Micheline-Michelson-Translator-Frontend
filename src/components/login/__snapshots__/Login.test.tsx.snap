// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches snapshot 1`] = `
"<Login>
  <div className=\\"login\\">
    <div className=\\"login-content\\">
      <Formik initialValues={{...}} validationSchema={{...}} onSubmit={[Function: submitForm]}>
        <form onSubmit={[Function]} noValidate={true}>
          <FormInput label=\\"Login\\" type=\\"text\\" name=\\"username\\" onChange={[Function]} onBlur={[Function]} value=\\"\\" errors={[undefined]} touched={[undefined]}>
            <div className=\\"form-input undefined\\">
              <p className=\\"form-input_label\\">
                Login
              </p>
              <input name=\\"username\\" type=\\"text\\" onChange={[Function]} onBlur={[Function]} placeholder=\\"Login\\" value=\\"\\" test-id=\\"input-test\\" className=\\"form-input_input\\" disabled={[undefined]} />
              <div>
                <div className=\\"form-input_validation-marker undefined\\" />
                <div className=\\"form-input_error\\" />
              </div>
            </div>
          </FormInput>
          <FormInput label=\\"Password\\" type=\\"password\\" name=\\"password\\" onChange={[Function]} onBlur={[Function]} value=\\"\\" errors={[undefined]} touched={[undefined]}>
            <div className=\\"form-input undefined\\">
              <p className=\\"form-input_label\\">
                Password
              </p>
              <input name=\\"password\\" type=\\"password\\" onChange={[Function]} onBlur={[Function]} placeholder=\\"Password\\" value=\\"\\" test-id=\\"input-test\\" className=\\"form-input_input\\" disabled={[undefined]} />
              <div>
                <div className=\\"form-input_validation-marker undefined\\" />
                <div className=\\"form-input_error\\" />
              </div>
            </div>
          </FormInput>
          <FormButton label=\\"Log in\\" type=\\"submit\\" disabled={false}>
            <button className=\\"form-button form-button_submit undefined\\" data-testid=\\"form-button\\" disabled={false} type=\\"submit\\" onClick={[undefined]}>
              Log in
            </button>
          </FormButton>
        </form>
      </Formik>
    </div>
  </div>
</Login>"
`;
